.github/workflows/CI-distro.yml

```diff
@@ -30,10 +30,6 @@ jobs:
         uses: julia-actions/setup-julia@v2
         with:
           version: '1'
-      - name: "Cache artifacts"
-        uses: julia-actions/cache@v2
```

Comment:
Are you sure this is an improvement? This caches the artifacts, so instead of downloading 160+ artifacts individually from Julia servers, this downloads a single cache file with all the artifacts.

It is difficult to compare individual CI runs on GitHub VMs, but the last ["CI 1.10 ubuntu" run on master](https://github.com/oscar-system/GAP.jl/actions/runs/13235926442/job/36940686024) took 4.5 minutes while [its counterpart here](https://github.com/oscar-system/GAP.jl/actions/runs/13242017330/job/36959330692) took 7 minutes.

Reply:
This is not related at all, since I am only touching the `distro CI` job, not the "normal" testing `CI`.

What this change essentially does is reducing the setup job from > 1 min (e.g. https://github.com/oscar-system/GAP.jl/actions/runs/13235926447/job/36940685206 last run on master) to ~15 sec (https://github.com/oscar-system/GAP.jl/actions/runs/13242017346/job/36959328839?pr=1156).